src/cli/update.php

```diff
@@ -309,12 +316,13 @@ public function updateCore()
 
 		$updateInformation = $jUpdate->getUpdateInformation();
 
-		if (!empty($updateInformation['object']))
+		if (!empty($updateInformation['hasUpdate']))
 		{
 			$packagefile = JInstallerHelper::downloadPackage($updateInformation['object']->downloadurl->_data);
-			$packagefile = JPATH_BASE . '/tmp/' . basename($packagefile);
-			$package = JInstallerHelper::unpack($packagefile, true);
-			JFolder::copy($package['extractdir'], JPATH_BASE, '', true, true);
+			$tmp_path    = $this->app->getCfg('tmp_path');
```

Comment:
I know, but it it will be removed in 4.0 and to the purpose of the downward compatibility...